services/notify/src/api.js

```diff
@@ -50,7 +66,19 @@ builder.declare({
   debug(`Received request to publish message on ${req.body.routingKey}`);
   await req.authorize({routingKey: req.body.routingKey});
   await this.notifier.pulse(req.body);
-  res.sendStatus(200);
+
+  let notificationAddress = {
+    notificationType: "pulse",
+    notificationAddress: req.body.routingKey,
+  };
+  // Ensure that the address is not in the blacklist
+  let response = await this.BlacklistedNotification.load(notificationAddress, true);
+  if(!response) {
+    await this.notifier.pulse(req.body);
+    res.sendStatus(200);
+  } else {
+    return res.reportError('BlacklistedAddress', `${req.body.address} is blacklisted`, {});
+  }
```

Comment:
Minor stylistic thing: it might be better to reverse the arms of this conditional.  Then the code to handle blacklisting (the `load` and the `reportError`) will be close together, and separated from the "normal path" of an address that is not blacklisted.